Editor.js as editor

carakas opened the pull request:
## Type

<!-- Remove the types that don't apply -->
<!-- If you discover any security related issues, please email core@fork-cms.com instead of using the issue tracker. -->

- Feature

## Resolves the following issues

<!-- List the hashes of the issues that this pull request resolves if their are issues for it. -->
<!-- Use the following format: fixes #[issue_number] -->
fixes #2945 

## Pull request description

<!-- Provide a summary of the pull request you are submitting. -->

Basic implementation of a block editor with paragraphs, headers, media library image and lists


https://editorjs.io/

jessedobbelaere commented:
@carakas I'm wondering, did you go for editorjs or sir trevor now? Because the diff seems to mention both in the code (package.json vs gulp tasks etc)? 👀

carakas commented:
@jessedobbelaere sorry, I started off with sir-trevor but then realised it's kinda abandoned, the implemented editor there is also deprecated so I switched. Must have forgotten some code, fixed in ee7b24dcc

carakas commented on src/Backend/Core/Js/editor/editor.js:
```diff
@@ -0,0 +1,70 @@
+      config[name].class = this.getClassFromVariableName(config[name].class)
+    }
+
+    this.create($element, config)
```
Yes

carakas commented on src/Common/BlockEditor/Twig/ParseBlocksExtension.php:
```diff
@@ -0,0 +1,78 @@
+        $cachedBlock = $cache->getItem($cacheKey);
+        if ($cachedBlock->isHit()) {
+            $parsedData = $cachedBlock->get();
+            if ($parsedData['time'] === $data['time'] && $parsedData['version'] === $data['version']) {
```
If we have a hit and something else is in there somebody has been tampering with the cache, I don't mind if it crashes in that case

carakas commented on src/Backend/Core/Js/backend.js:
```diff
@@ -565,6 +569,34 @@ jsBackend.ckeditor = {
+  },
+
+  loadEditorsInCollections: function () {
+    $('[data-addfield=collection]').on('collection-field-added', function () {
```
fixed in fda209032

carakas commented on src/Common/BlockEditor/Blocks/AbstractBlock.php:
```diff
@@ -0,0 +1,51 @@
+    final protected function parseWithTwig(string $template, array $data): string
+    {
+        /** @var TwigTemplate $templating */
+        $templating = $this->container->get('templating');
```
Fixed in a07e53fe5
I think initially something didn't work but with the current implementation it works
